src/components/ConnectButton/index.tsx

```diff
@@ -11,20 +12,21 @@ export interface ConnectButtonProps
 }
 
 export const ConnectButton = (props: ConnectButtonProps) => {
-  const { connected, connect, select, provider } = useWallet();
+  const { connected, connect, select, wallet } = useWallet();
   const { onClick, children, disabled, allowWalletChange, ...rest } = props;
 
   // only show if wallet selected or user connected
-
   const menu = (
     <Menu>
-      <Menu.Item key="3" onClick={select}>
-        Change Wallet
-      </Menu.Item>
+      {Object.entries(WalletName).map(([key, value]) => (
+        <Menu.Item key="3" onClick={() => select(key as WalletName)}>
```

Comment:
Looping over this enum is incorrect. Use the `wallets` value provided by `useWallet` to get the list of wallets injected into the context by the app.

Components rendered from an array like this need to have unique keys.

`key as WalletName` is not a valid way to use an enum value. If you have to cast something, consider that the type you're using needs to be changed.

Reply:
Good catch! Alright I was able to fix it, now I know what this piece does.